sql/src/main/java/io/crate/metadata/sys/SysSnapshotsTableInfo.java

```diff
@@ -21,73 +21,29 @@
 import io.crate.metadata.ColumnIdent;
 import io.crate.metadata.RelationName;
-import io.crate.metadata.Routing;
-import io.crate.metadata.RoutingProvider;
-import io.crate.metadata.RowGranularity;
-import io.crate.metadata.expressions.RowCollectExpressionFactory;
-import io.crate.metadata.table.ColumnRegistrar;
-import io.crate.metadata.table.StaticTableInfo;
-import org.elasticsearch.cluster.ClusterState;
-
-import java.util.Map;
-
-import static io.crate.execution.engine.collect.NestableCollectExpression.forFunction;
+import io.crate.metadata.SystemTable;
 import static io.crate.types.DataTypes.STRING;
 import static io.crate.types.DataTypes.STRING_ARRAY;
 import static io.crate.types.DataTypes.TIMESTAMPZ;
 
-public class SysSnapshotsTableInfo extends StaticTableInfo<SysSnapshot> {
+public class SysSnapshotsTableInfo {
 
     public static final RelationName IDENT = new RelationName(SysSchemaInfo.NAME, "snapshots");
-    private static final RowGranularity GRANULARITY = RowGranularity.DOC;
-
-    public static class Columns {
-        public static final ColumnIdent NAME = new ColumnIdent("name");
-    }
-
-    static Map<ColumnIdent, RowCollectExpressionFactory<SysSnapshot>> expressions() {
-        return columnRegistrar().expressions();
-    }
-
-    private static ColumnRegistrar<SysSnapshot> columnRegistrar() {
-        return new ColumnRegistrar<SysSnapshot>(IDENT, GRANULARITY)
-          .register("name", STRING, () -> forFunction(SysSnapshot::name))
-          .register("repository", STRING, () -> forFunction(SysSnapshot::repository))
-          .register("concrete_indices", STRING_ARRAY, () -> forFunction(SysSnapshot::concreteIndices))
-          .register("started", TIMESTAMPZ, () -> forFunction(SysSnapshot::started))
-          .register("finished", TIMESTAMPZ, () -> forFunction(SysSnapshot::finished))
-          .register("version", STRING, () -> forFunction(SysSnapshot::version))
-          .register("state", STRING, () -> forFunction(SysSnapshot::state))
-          .register("failures", STRING_ARRAY, () -> forFunction(SysSnapshot::failures));
-    }
-
-    SysSnapshotsTableInfo() {
-        super(IDENT, columnRegistrar(), "name","repository");
-    }
-
-    @Override
-    public RowGranularity rowGranularity() {
-        return GRANULARITY;
-    }
-
-    @Override
-    public RelationName ident() {
-        return IDENT;
-    }
 
-    @Override
-    public Routing getRouting(ClusterState clusterState,
-                              RoutingProvider routingProvider,
-                              WhereClause whereClause,
-                              RoutingProvider.ShardSelection shardSelection,
-                              SessionContext sessionContext) {
-        // route to random master or data node,
-        // because RepositoriesService (and so snapshots info) is only available there
-        return routingProvider.forRandomMasterOrDataNode(IDENT, clusterState.getNodes());
```

Comment:
This custom routing behavior needs to be preserved.

Reply:
👍